core/state_transition.go

```diff
@@ -650,6 +660,10 @@ func (st *stateTransition) innerExecute() (*ExecutionResult, error) {
 				}
 				st.state.AddBalance(params.OptimismL1FeeRecipient, amtU256, tracing.BalanceIncreaseRewardTransactionFee)
 			}
+			if rules.IsOptimismIsthmus {
+				operatorFeeCost := st.evm.Context.OperatorCostFunc(st.gasUsed(), st.evm.Context.Time)
```

Comment:
Note that the value of `st.gasUsed()` will potentially be smaller here than in the sender operator fee refund calculation inside `refundGas` because it already got state refunds subtracted after the sender op fee refund. This also means that the refunded gas to the `msg.Sender` is potentially less than if we refunded after applying state refunds.

This may be intended, but I'm not sure. I believe it is currently underspecified in the specs and we should clarify.

Reply:
The latest rebase changed the behavior to now include the state refund in the operator fee refund. So the user refund and operator fee to the operator fee refund seem now consistent.

Reply:
I've confirmed with @clabby that this aligns with the Rust implementation now. State refunds are applied before operator cost refund. This needs to be added to the specs.